contracts/SinglePlayerCommit.sol

```diff
@@ -51,213 +61,470 @@ contract SinglePlayerCommit is Ownable {
+    /// @param _stake Amount of <token> to stake againt achieving goale
+    /// @param _depositAmount Size of deposit
+    /// @param _userId ???
+    /// @dev Call deposit and makeCommitment method
+    function depositAndCommit(
+        bytes32 _activityKey,
+        uint256 _goalValue,
+        uint256 _startTime,
+        uint256 _stake,
+        uint256 _depositAmount,
+        string memory _userId
+    ) 
+        public 
+        returns (bool success) 
+    {
+        require(deposit(_depositAmount), "SPC::depositAndCommit - deposit failed");
+        require(makeCommitment(
+                    _activityKey, 
+                    _goalValue, 
+                    _startTime, 
+                    _stake, 
+                    _userId
+                ), "SPC::depositAndCommit - commitment creation failed");
 
-    // TODO
-    function report() public returns (bool) {
-        // get activity data from oracle
-        // record activity data in commitments
         return true;
     }
 
+    //TODO DRY in procesCommitment methods
+    /// @notice Enables processing of open commitments after endDate that have not been processed by creator
+    /// @param committer address of the creator of the committer to process
+    /// @dev Process commitment by lookup based on address, checking metrics, state and updating balances
     function processCommitment(address committer) public {
         console.log("Processing commitment");
-        Commitment memory commitment = commitments[committer];
+        Commitment storage commitment = commitments[committer];
 
-        // check if commitment has ended
         require(commitment.endTime < block.timestamp, "SPC::processCommitment - commitment is still active");
+        require(commitment.endTime < commitment.lastActivityUpdate, "SPC::processCommitment - update activity");
 
-        bool met = commitment.met;
-        uint256 stake = commitment.stake;
+        commitment.met = commitment.reportedValue > commitment.goalValue;
 
-        // "delete" the expired commitment
-        commitments[committer].exists = false;
+        if (!commitment.met) {
+            _slashFunds(commitment.stake, committer);
+        } 
+        
+        commitment.exists = false;
+        emit CommitmentEnded(committer, commitment.met, commitment.stake);
+    }
 
-        uint256 penalty;
+    /// @notice Enables control of processing own commitment. For instance when completed.
+    /// @dev Process commitment by lookup msg.sender, checking metrics, state and updating balances
+    function processCommitmentUser() public {
+        console.log("Processing commitment");
+        Commitment storage commitment = commitments[msg.sender];
 
-        if (met) {
-            penalty = 0;
-        } else {
-            penalty = stake;
-            // remove from committer's balance
-            _changeCommitterBalance(penalty, false);
-        }
+        commitment.met = commitment.reportedValue > commitment.goalValue;
```

Comment:
let's factor the common logic from `processCommitment` and `processCommitmentUser` out into its own (internal?) function

Reply:
This entrails that processing commitment has 2 possible subsequent calls, first to the common logic, than to the slashin function. I believe you mentioned something about not adding to much method calls? That's why I went for this approach, so are we sure?

Reply:
I stand corrected: https://docs.soliditylang.org/en/v0.5.4/control-structures.html#internal-function-calls

So I'll add it as an internal method

Reply:
cool, thanks

Reply:
done

Reply:
nice! 

minor naming change: `_settleCommitment` instead of `_commitmentSettlement` is more of a verb/method/function